gui_qt/litellm_worker.py

```
@@ -50,32 +68,115 @@ def _http_error_message(exc: HTTPError, label: str) -> str:
+    def _start_budget(self, total_seconds: float) -> None:
+        self._deadline = time.monotonic() + max(0.0, float(total_seconds))
+
+    def _remaining_timeout(self, *, cap: float | None = None) -> float:
+        """Return urlopen timeout for the next hop within the shared budget."""
+        per_request = float(CATALOG_TIMEOUT_SECONDS if cap is None else cap)
+        if self._deadline is None:
+            return per_request
+        remaining = self._deadline - time.monotonic()
+        if remaining < MIN_REQUEST_TIMEOUT_SECONDS:
+            raise BudgetExhausted(
+                f"联网加载总时限（约 {CATALOG_TOTAL_BUDGET_SECONDS:g} 秒）已用尽"
```

Ruff (0.16.0)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)